src/main/python/model/postbuilder.py

```diff
@@ -0,0 +1,117 @@
+
+from model.iir import Gain
+from model.minidsp import HDXmlParser, pad_with_passthrough
+from model.preferences import BEQ_DOWNLOAD_DIR
+from ui.postbuilder import Ui_postbuilder
+
+logger = logging.getLogger('postbuilder')
+
+
+class CreatePostDialog(QDialog, Ui_postbuilder):
+    '''
+    Create Post dialog
+    '''
+
+    def __init__(self, parent, prefs, filter_model):
+        super(CreatePostDialog, self).__init__(parent)
+        self.setupUi(self)
+        self.__preferences = prefs
+        self.__beq_dir = self.__preferences.get(BEQ_DOWNLOAD_DIR)
+        self.__filter_model = filter_model
+
+    def generate_post(self):
+        '''
+        Creates the output content.
+        '''
+        metadata = self.__build_metadata()
+        audio_display = ' / '.join(metadata['beq_audioTypes'])
+
+        season_display = ''
+        source_display = ''
+        post_warning = ''
+
+        save_name = f"{metadata['beq_title']} ({metadata['beq_year']})"
+
+        if len(metadata['beq_season']) > 0:
+            season_display = f"Season {metadata['beq_season']}"
+            save_name += f' ({season_display})'
+
+        if metadata['beq_source'] != 'Disc':
+            source_display = metadata['beq_source']
+            save_name += f" ({metadata['beq_source']})"
+
+        if len(metadata['beq_warning']) > 0:
+            post_warning = f'\n[SIZE="3"][COLOR="DarkRed"][B]WARNING: {metadata["beq_warning"]}[/B][/COLOR][/SIZE]\n\n'
+
+        post = f"[CENTER][B]BassEQ {metadata['beq_title']} ({metadata['beq_year']}) {season_display} {metadata['beq_edition']} {source_display} {audio_display}[/B][/CENTER]\n{post_warning}[IMG]{metadata['beq_pvaURL']}[/IMG]\n[IMG]{metadata['beq_spectrumURL']}[/IMG]"
+
+        self.postTextEdit.clear()
+        self.postTextEdit.insertPlainText(post)
+
+        gain_filter = self.__find_gain(self.__filter_model.filter)
+
+        if gain_filter is not None:
+            save_name += f' ({gain_filter.gain:+.1f} gain)'
+            metadata['beq_gain'] = f'{gain_filter.gain:+.1f}'
+
+        save_name += f' BEQ {metadata["beq_audioTypes"][0]}'
+        save_name = save_name.replace(':', '-')
+
+        file_name = QFileDialog(self).getSaveFileName(self, 'Export BEQ Filter', f'{save_name}.xml', 'XML (*.xml)')
+        file_name = str(file_name[0]).strip()
+
+        if len(file_name) > 0:
+            if getattr(sys, 'frozen', False):
+                file_path = os.path.join(sys._MEIPASS, 'flat24hd.xml')
+            else:
+                file_path = os.path.abspath(os.path.join(os.path.dirname('__file__'), '../xml/flat24hd.xml'))
+            filters = pad_with_passthrough(self.__filter_model.filter, 96000, 10)
+            output_xml = HDXmlParser('2x4 HD').overwrite(filters, file_path, metadata)
+            with open(file_name, 'w') as f:
+                f.write(output_xml)
+
+    def __build_metadata(self):
+        metadata = {'beq_title': self.titleField.text()}
```

Comment:
you can write this as a dict literal

Reply:
Is there a reason to do this in Python? Performance or something?  PyCharm had made the suggestion too.  I'm used to it mainly being a style thing in compiled languages.  I usually go with this form over a dict literal when there's more than a couple items because I find it easier to read and edit. I'm cool to change it.  The question is out of curiosity and learning.

Reply:
it's a preference thing unless you're doing this in a tight loop (in which case there is a small but measurable performance difference afaik) 

arguably writing it like 

    foo = {
        'bar': some_var,
        'baz': another_var    
    }

is more idiomatic though

same as 

    foo = [bar,baz]

instead of

    foo = list()
    foo.append(bar)
    foo.append(baz)

yours is actually a mix of both styles as it is a dict literal which is then extended (as opposed to using dict() and then putting stuff into it)

Reply:
Ah, cool.  Thanks for the info.  Mine is how it's commonly done in objc and swift which is my day job.  I'll keep it in mind for python code.